src/voting/CIP-voting.md

```diff
@@ -0,0 +1,166 @@
+---
+CIP: 62
+Title: Cardano dApp-Connector Governance extension
+Authors: Bruno Martins <bruno.martins@iohk.io>
+Status: Draft
+Type: Standards
+Created: 2021-06-11
+License: CC-BY-4.0
+---
+
+# **Abstract**
+
+This document describe the interface between webpage / web-based stack and cardano wallets. This specificies that API of the javascript object that need to be injected into the web applications in order to support all the Governance features.
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+
+# **Motivation**
+The goal for this CIP is to extend the dApp-Wallet web bridge to enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+# **Specification**
+
+## `Types`
+
+### **GovernanceKey**
+
+```
+type GovernanceKey = {
+  votingKey: string,
+  weight: number
+}
+
+```
+
+`votingKey`: Ed25519 pubkey 32 bytes HEX string  
+
+`weight`: Used to calculate the actual voting power using the rules described
+in 
+[CIP-36](https://cips.cardano.org/cips/cip36/).
+
+
+### **Purpose**
+
+```
+type enum Purpose = {
+  CATALYST = 0,
+  OTHER = 1
+}
+
+```
+
+`Purpose`: Defines the purpose of the delegations. This is used to limit the scope of the delegations.  For example, a purpose might be a subset of Catalyst proposals, a council election, or even some private purpose (agreed by convention).
+
+## **Namespace**
+
+### **cardano.{walletName}.governance.enable(): Promise\<API>**
+The `cardano.{walletName}.governance.enable()` method is used to enable the governance API. It should request permission from the wallet to enable the API. If permission is granted, the rest of the API will be available. The walelt may choose to maintain a whitelist of allowed clients to avoid asking for permission every time.
+
+This api being an extension of [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/), expects that `cardano.{walletName}.enable()` to be enabled implicitly. 
+
+# **`API`**
+
+## **api.getVotingKey**(address_index: number = 0, account: number = 0, role: number = 0): Promise\<Bip32PublicKey>
```

Comment:
Bip32PublicKey is the extended one (can derive from it). Just double-checking if we really need it here, because **Returns** section says it's just a
> `cbor\<vkey>` - cbor serialized 64 bytes (x, y) Ed25519 public key